src/analysis/inner_product_space/adjoint.lean

```diff
@@ -368,10 +338,112 @@ lemma is_adjoint_pair_inner (A : E' →L[ℝ] F') :
 
 end real
 
+/-! ### Self-adjoint operators -/
+
+/-- An operator `A` on an inner product space is self-adjoint iff `A` is equal to its adjoint.
+
+TODO Extend this theory to unbounded operators. -/
+def is_self_adjoint (A : E →L[𝕜] E) : Prop := A.adjoint = A
```

Comment:
Maybe I'm misunderstanding the upshot of our discussion on Zulip, but weren't we going to migrate this to the context of `[ring R] [star_ring R] (A : R)` rather than `(A : E →L[𝕜] E)` before it is merged?

Reply:
Oh! I hadn't realised this was supposed to happen as part of this. Either now or later would be fine with me.

Reply:
It's new material in this PR though, not a reformulation of existing material.  So I'd have thought we should decide on the right context before merging?  (cc @j-loreaux also for thoughts)

Reply:
You're absolutely right Heather I was probably too quick in my assessment. Thank you for giving this PR the close attention that it deserves.

I am still confident in Jireh's judgement and I am sure he will have no objections to waiting till you have given your blessing to the final form.

Reply:
Yes, I'm fine with waiting. I have been waiting in part because I'm currently on vacation, so haven't had time to look at the code since the Zulip discussion.